Fix flaky recreate_index test

We have another flaky celery task test on Travis: `search.tasks_test.test_start_recreate_index`. It seems to only happen on the 'push' build and not the 'pr' build.

The fix for a different flaky celery test (#1679) will probably be relevant here.

### Failure output:
```
=================================== FAILURES ===================================
__________________________ test_start_recreate_index ___________________________
mock_index_functions = namespace(new_profile=<MagicMock name='index_new_profile' id='139707847508824'>, update_author=<MagicMock name='update_author' id='139707847525488'>, update_posts=<MagicMock name='update_author_posts_comments' id='139707847021848'>)
mocker = <pytest_mock.MockFixture object at 0x7f10449eee80>
mocked_celery = namespace(chain=<MagicMock name='chain' spec='chain' id='139707846829392'>, group=<MagicMock name='group' spec='group' id='139707847052032'>, replace=<function replace at 0x7f10449e7ae8>, replace_exception_class=<class 'TabError'>)
user = <User: 01D19JF6YP0F946A1Q2CSTEXH9>
    def test_start_recreate_index(mock_index_functions, mocker, mocked_celery, user):
        """
        recreate_index should recreate the elasticsearch index and reindex all data with it
        """
        settings.INDEXING_API_USERNAME = user.username
        settings.ELASTICSEARCH_INDEXING_CHUNK_SIZE = 2
        users = UserFactory.create_batch(4)
        channels = [ChannelFactory.create(name=name) for name in ["a", "b", "c"]]
        index_channel_mock = mocker.patch("search.tasks.index_channel", autospec=True)
        index_profiles_mock = mocker.patch("search.tasks.index_profiles", autospec=True)
        backing_index = "backing"
        create_backing_index_mock = mocker.patch(
            "search.indexing_api.create_backing_index",
            autospec=True,
            return_value=backing_index,
        )
        finish_recreate_index_mock = mocker.patch(
            "search.tasks.finish_recreate_index", autospec=True
        )
    
        with pytest.raises(mocked_celery.replace_exception_class):
            start_recreate_index.delay()
        for doctype in VALID_OBJECT_TYPES:
            create_backing_index_mock.assert_any_call(doctype)
        finish_recreate_index_mock.s.assert_called_once_with(
            {"post": backing_index, "comment": backing_index, "profile": backing_index}
        )
        assert mocked_celery.group.call_count == 1
    
        # Celery's 'group' function takes a generator as an argument. In order to make assertions about the items
        # in that generator, 'list' is being called to force iteration through all of those items.
        list(mocked_celery.group.call_args[0][0])
        for channel in channels:
            index_channel_mock.si.assert_any_call(channel.id)
            mocked_celery.chain.assert_called_once_with(
                mocked_celery.group.return_value, finish_recreate_index_mock.s.return_value
            )
>       index_profiles_mock.si.assert_any_call([users[0].profile.id, users[1].profile.id])
E       AssertionError: si([1637, 1638]) call not found
E       
E       pytest introspection follows:
E       
E       Args:
E       assert ([1640, 1637],) == ([1637, 1638],)
E         At index 0 diff: [1640, 1637] != [1637, 1638]
E         Use -v to get the full diff
search/tasks_test.py:209: AssertionError
----------------------------- Captured stderr call -----------------------------
[2019-01-15 20:15:31] INFO 44 [search.tasks] tasks.py:167 - [15bd906684f8] - starting to index all channels, posts, comments, and profiles...
------------------------------ Captured log call -------------------------------
```

--- a/search/tasks.py
+++ b/search/tasks.py
@@ -176,6 +176,7 @@ def start_recreate_index(self):
                 for ids in chunks(
                     User.objects.exclude(username=settings.INDEXING_API_USERNAME)
                     .exclude(profile__isnull=True)
+                    .order_by("id")
                     .values_list("profile__id", flat=True),
                     chunk_size=settings.ELASTICSEARCH_INDEXING_CHUNK_SIZE,
                 )

--- a/search/tasks_test.py
+++ b/search/tasks_test.py
@@ -175,7 +175,7 @@ def test_start_recreate_index(mock_index_functions, mocker, mocked_celery, user)
     """
     settings.INDEXING_API_USERNAME = user.username
     settings.ELASTICSEARCH_INDEXING_CHUNK_SIZE = 2
-    users = UserFactory.create_batch(4)
+    users = sorted(UserFactory.create_batch(4), key=lambda user: user.id)
     channels = [ChannelFactory.create(name=name) for name in ["a", "b", "c"]]
     index_channel_mock = mocker.patch("search.tasks.index_channel", autospec=True)
     index_profiles_mock = mocker.patch("search.tasks.index_profiles", autospec=True)

